lib/types.js

```diff
@@ -1246,38 +1301,33 @@ UnionType.prototype._branchConstructor = function () {
  * + `map`, `record`
  */
 class UnwrappedUnionType extends UnionType {
-  constructor (schema, opts) {
+  /**
+   *
+   * @param {*} schema
+   * @param {*} opts
+   * @param {Function|undefined} projectionFn The projection function used
+   *                                          to determine the bucket for the
+   *                                          Union. Falls back to generate
+   *                                          from `wrapUnions` parameter
+   *                                          if given.
+   */
```

Comment:
Let's delete this comment, the third argument is an implementation detail and should not be considered part of the public API.

Reply:
removed in 0e6c4cd; marked the param with an underscore prefix and a jsdoc-style inline comment, so it shows at least up in intellisense - I think the private parameter is confusing at best. I think writing it on the object before when the union type is determined or possibly using a `Symbol` in opts would be better.

Reply:
It's not ideal, but better than mutating the options (which may be reused to surprising results, etc.). Once `avsc` is using TypeScript the constructor can be hidden from users of the library altogether.